web-runtime/npm-package/src/FluenceAppService.ts

```diff
@@ -0,0 +1,119 @@
+import { FaaSConfig, Envs } from './config';
+import { IFluenceAppService } from './types';
+import { isBrowser, isNode } from 'browser-or-node';
+import { Thread, ModuleThread, spawn, Worker } from 'threads';
+
+export const defaultNames = {
+    avm: {
+        name: 'avm.wasm',
+        package: '@fluencelabs/avm',
+    },
+    marine: {
+        name: 'marine-js.wasm',
+        package: '@fluencelabs/marine-js',
+    },
+    script: {
+        web: './marine-js.web.js',
+        node: './marine-js.node.js',
+    },
+};
+
+export const bufferToSharedArrayBuffer = (buffer: Buffer): SharedArrayBuffer => {
+    const sab = new SharedArrayBuffer(buffer.length);
+    const tmp = new Uint8Array(sab);
+    tmp.set(buffer, 0);
+    return sab;
+};
+
+export const loadWasmFromUrl = async (fileName: string): Promise<Buffer> => {
```

Comment:
Not sure if the Url part of the name is the best. The function downloads wasm not from an arbitrary URL, but from current site. I think that more specific name will be better, but nothing comes to my mind right now